MD5 Checksum changes (#81329)
openshift/openshift-docs: modules/oadp-configuring-aws-md5sum.adoc

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc

:_mod-docs-content-type: PROCEDURE
[id="oadp-configuring-aws-md5sum_{context}"]
= Configuring the backup storage location with MD5 checksum algorithm

> You can configure the Backup Storage Location (BSL) in the Data Protection Application (DPA) to use a MD5 checksum algorithm for {aws-short} S3 and S3 compatible storage providers. The checksum algorithm calculates the checksum for uploading and downloading objects to S3. You can use one of the following options to set the `checksumAlgorithm` field in the `spec.backupLocation.velero.config.checksumAlgorithm` section of the DPA.

Reviewer: ```suggestion
You can configure the Backup Storage Location (BSL) in the Data Protection Application (DPA) to use a MD5 checksum algorithm for {aws-short} S3 and S3 compatible storage providers. The checksum algorithm calculates the checksum for uploading and downloading objects to S3. You can use one of the following options to set the `checksumAlgorithm` field in the `spec.backupLocations.velero.config.checksumAlgorithm` section of the DPA.
```

Reviewer 2: +1

> You can configure the Backup Storage Location (BSL) in the Data Protection Application (DPA) to use a MD5 checksum algorithm for {aws-short} S3 and S3 compatible storage providers. The checksum algorithm calculates the checksum for uploading and downloading objects to S3. You can use one of the following options to set the `checksumAlgorithm` field in the `spec.backupLocations.velero.config.checksumAlgorithm` section of the DPA.

Reviewer 3: ```suggestion
You can configure the Backup Storage Location (BSL) in the Data Protection Application (DPA) to use a MD5 checksum algorithm for both Amazon Simple Storage Service (Amazon S3) and S3-compatible storage providers. The checksum algorithm calculates the checksum for uploading and downloading objects to Amazon S3. You can use one of the following options to set the `checksumAlgorithm` field in the `spec.backupLocations.velero.config.checksumAlgorithm` section of the DPA.
```
Hm, it appears that Amazon calls this "Amazon Simple Storage Service (Amazon S3)", see https://aws.amazon.com/s3/

* `CRC32`
* `CRC32C`
* `SHA1`
* `SHA256`

[NOTE]
====
You can also set the `checksumAlgorithm` field to an empty value to skip the MD5 checksum check. 

If you do not set a value for the `checksumAlgorithm` field, then the default value is set to `CRC32`.
====

.Prerequisites

* You have installed the {oadp-short} Operator.

> * You have configured {aws-short} S3, or S3 compatible object storage as a backup location.

Reviewer 3: Here as well ensure that we're referring to the correct thing.

Author: Done. thanks.

.Procedure

* Configure the BSL in the DPA as shown in the following example:
+
.Example DPA
[source,yaml]
----
apiVersion: oadp.openshift.io/v1alpha1
kind: DataProtectionApplication
metadata:
  name: test-dpa
  namespace: openshift-adp
spec:
  backupLocations:
  - name: default
    velero:
      config:
        checksumAlgorithm: "" # <1>
        insecureSkipTLSVerify: "true"
        profile: "default"
        region: <bucket_region> 
        s3ForcePathStyle: "true"
        s3Url: <bucket_url>
      credential:
        key: cloud
        name: cloud-credentials
      default: true
      objectStorage:
        bucket: <bucket_name> 
        prefix: velero
      provider: aws
  configuration:
    velero:
      defaultPlugins:
      - openshift
      - aws
      - csi
----
<1> Specify the `checksumAlgorithm`. In this example, the `checksumAlgorithm` field is set to an empty value. You can select an option from the following list: `CRC32`, `CRC32C`, `SHA1`, `SHA256`.

[IMPORTANT]
====

> If you are using Noobaa as the object storage provider and you do not set the `spec.backupLocation.velero.config.checksumAlgorithm` field in the DPA, an empty value of `checksumAlgorithm` is added to the BSL configuration. 

Reviewer 4: ```suggestion
If you are using Noobaa as the object storage provider, and you do not set the `spec.backupLocation.velero.config.checksumAlgorithm` field in the DPA, an empty value of `checksumAlgorithm` is added to the BSL configuration. 
```
